sklearn/manifold/isomap.py

```diff
@@ -196,16 +195,192 @@ def transform(self, X):
+
+    random_state : int, optional (default = None)
+        The seed for the random landmark selector.
+        This parameter will be ignored if landmarks was passed.
+
+    Attributes
+    ----------
+    embedding_ : array-like, shape (n_samples, n_components)
+        Stores the embedding vectors.
+
+    kernel_pca_ : object
+        `KernelPCA` object used to implement the embedding.
+
+    training_data_ : array-like, shape (n_samples, n_features)
+        Stores the training data.
+
+    nbrs_ : sklearn.neighbors.NearestNeighbors instance
+        Stores nearest neighbors instance, including BallTree or KDtree
+        if applicable.
+
+    dist_matrix_ : array-like, shape (n_samples, n_samples)
+        Stores the geodesic distance matrix of training data.
+
+    landmarks_ : array-like, shape (n_landmarks,)
+        Stores the indices of the selected landmarks.
+
+    References
+    ----------
+
+    .. [1] Silva, V. D., & Tenenbaum, J. B. (2002). Global versus local
+           methods in nonlinear dimensionality reduction. In Advances
+           in neural information processing systems (pp. 705-712).
+    """
+
+    def __init__(self, n_neighbors=5, n_components=2, eigen_solver='auto',
+                 tol=0, max_iter=None, path_method='auto',
+                 neighbors_algorithm='auto',
+                 n_landmarks='auto', landmarks=None,
+                 random_state=None):
+        super(LandmarkIsomap, self).__init__(
+            n_neighbors, n_components, eigen_solver, tol, max_iter,
+            path_method, neighbors_algorithm)
+
+        self.random_state = random_state
+        self.n_landmarks = n_landmarks
+        self.landmarks_ = landmarks
+
+    def _get_landmarks(self, X):
+        """Get the landmarks from a data set X.
+
+        If the landmarks were passed in the constructor, returns them.
+        Otherwise, randomly selects n_landmarks (contained in the
+        interval [0, |X|)).
+
+        If n_landmarks is 'auto', selects n_components + 1 (the necessary
+        landmarks required to triangulate a samples' position in the
+        n_components-dimensional embedding plus a security margin (1% of
+        the size of the data set X).
+
+        Parameters
+        ----------
+        X
+            The data set containing the samples from where the landmarks
+            should be inferred.
+
+        Returns
+        -------
+        landmarks: array-like, shape (n_landmarks,)
+        """
+        if self.landmarks_ is None:
+            n_samples = X.shape[0]
+            n_landmarks = self.n_landmarks
+
+            if n_landmarks == 'auto':
+                # To guarantee some stability, the number of landmarks
+                # should be strictly superior to the number of dimensions
+                # of the reduced data set added by some safety margin.
+                n_landmarks = self.n_components + 1 \
+                              + min(self.n_components // 2, 100)
```

Comment:
Style thing: I'd go for line wrapping in parentheses rather than with a slash-escape.